File: day1-part1.hs
module Main where

import System.IO
import Data.List

main :: IO ()
main = do
    input <- readFile "day1-input.txt"
    let
        pairs = parseInput input
        (left, right) = unzip pairs
        sortedLeft = sort left
        sortedRight = sort right
        result = sum $ zipWith diff sortedLeft sortedRight

    putStrLn $ show result

diff x y = abs (x-y)
parseInput input = toListTupleInt $ toListTuple $ splitPair $ breakLine input
breakLine input = takeWhile (\l -> l /= "") $ lines input
splitPair lines = map (\l -> words l) lines
toListTuple lines = map (\[x,y] -> (x,y)) lines
toListTupleInt lines = map (\(x,y) -> (read x :: Int, read y :: Int)) lines

File: day1-part1.rs
fn main() {
    let input = include_str!("day1-input.txt");
    let mut sum = 0;
    let mut left: Vec<i32> = Vec::new();
    let mut right: Vec<i32> = Vec::new();
    let s: Vec<Vec<&str>> = input.lines().map(|l| l.split("   ").collect::<Vec<&str>>()).collect();
    println!("{:?}", s);
    // for line in input.lines() {
    //     let words: Vec<&str> = line.split("   ").collect();
    //     for w in words {
    //         let x = w.parse::<i32>().unwrap();
    //         left.push(x);
    //         right.push(x);
    //     }
    // }
    // println!("left {:?}", left);
    // println!("right {:?}", right);
}
